openpathsampling/tests/analysis/utils/mock_movers.py

```diff
@@ -0,0 +1,464 @@
+"""
+Tools for generating sequences of steps for analysis tests.
+
+The idea here is to have client code specify things like which snapshot
+index is the shooting point and what partial trajectory from one-way
+shooting should be, then we patch those things and use the internals of the
+mover to generate the real data.
+
+In this way, we use as much of the actual OPS machinery as possible,
+ensuring, for example, that the details we return are correct.
+
+Note that for the classes here, each instance represents a single move
+(single MC step). This is unlike the PathMover objects in OPS, which
+represents an individual type of move that can be reused for many steps.
+"""
+
+import random
+try:
+    from unittest.mock import Mock, patch
+except ImportError:  # pragma: no cover
+    from mock import Mock, patch
+
+import openpathsampling as paths
+
+
+def _get_only(iterable, condition, error_msg):
+    possibilities = [item for item in iterable if condition(item)]
+    if len(possibilities) != 1:
+        msg = "expected 1 {error_msg}; found {nfound}: {found}".format(
+            error_msg=error_msg,
+            nfound=len(possibilities),
+            found=possibilities
+        )
+        raise AnalysisTestSetupError(msg)
+    return possibilities[0]
+
+
+class AnalysisTestSetupError(Exception):
+    """Raised for when an internal error occurs during test setup.
+
```

Comment:
hard to read grammar
```suggestion
    """Raised when an internal error occurs during test setup.

```